### PR TITLE
fix crash in <PHP8 with ResourceBundle

### PR DESCRIPTION
**Pull request by kkmuffme**
Fix https://github.com/vimeo/psalm/issues/8346

partially reverts https://github.com/vimeo/psalm/pull/8217

### CODE REVIEW

**Comment by orklah** [comment]
count used to contain ResourceBundle, the previous PR removed it for newer version. I don't think dropping it altogether is the way to go here. We better keep it even on PHP >8

**Comment by orklah** [comment]
count used to have ResourceBundle: https://github.com/vimeo/psalm/pull/8217/files#diff-5611cd00854405fe61befddd211dacd225d9c479b499ffd7e3c0c7d96eb568edR216

The change in https://github.com/vimeo/psalm/pull/8217 was to remove ResourceBundle in latest php versions definitions. So reverting that would mean still have ResourceBundle on every version like before, not removing it altogether

**Comment by AndrolGenhald** [comment]
FWIW I do think `ResourceBundle` needs removed from the callmap (see [here](https://github.com/vimeo/psalm/issues/8346#issuecomment-1216940311)). Referencing extension-defined classes in the callmap in functions not defined by that extension is probably a bad idea. I'm not sure what the best fix is though, a provider for `count` seems like overkill in this case, but that might be the only way to correctly fix it.

**Comment by orklah** [comment]
ouch, I didn't know it was an extension... Let's fix this the way it was and we'll look for a solution after I think

@kkmuffme can you check tests 1 and 4? There's callmap consistencies failures in tests

File: tests/Internal/Codebase/InternalCallMapHandlerTest.php [hunk 1]
```diff
@@ -519,7 +519,13 @@ public function testIgnoresAreSortedAndUnique(): void
             /** @var string */
             $function = is_int($key) ? $value : $key;
 
-            $this->assertGreaterThan(0, strcmp($function, $previousFunction));
+            $diff = strcmp($function, $previousFunction);
```
Comment (orklah) [inline tests/Internal/Codebase/InternalCallMapHandlerTest.php:522]: Is this on purpose? Won't PHPUnit scream if there are outputs in a test?

**Comment by orklah** [comment]
Thanks!

**Pull request merged**